.github/workflows/docker.yml

```
@@ -0,0 +1,118 @@
+name: Docker
+on: [workflow_dispatch, push]
```

```suggestion
on: [workflow_dispatch, pull_request]
```
I think using `pull_request` here should trigger the action in the PR and we can try all of this in isolation from the `main` branch. I am not sure if having both means it is a job that needs manual trigger additionally to a PR, but that would be favourable.

---

the `workflow_dispatch` is useful whenever you need to trigger the workflow manually, but it's not required for when a `pull_request` is created, thus being just an additional (handy) option :)